packages/components/button/src/button.tsx

```diff
@@ -21,8 +21,20 @@ const Button = forwardRef<"button", ButtonProps>((props, ref) => {
     disableRipple,
     getButtonProps,
     getRippleProps,
+    isIconOnly,
   } = useButton({...props, ref});
 
+  if (isIconOnly && isLoading) {
+    return (
+      <Component ref={domRef} className={styles} {...getButtonProps()}>
+        {startContent}
+        {isLoading && spinner}
+        {endContent}
+        {!disableRipple && <Ripple {...getRippleProps()} />}
+      </Component>
+    );
+  }
+
```

Comment:
Hey @qWici thanks!, instead of repeating the same logic as the return block try doing this:

```suggestion
 - if (isIconOnly && isLoading) {
 -   return (
 -     <Component ref={domRef} className={styles} {...getButtonProps()}>
 -       {startContent}
 -       {isLoading && spinner}
 -       {endContent}
 -       {!disableRipple && <Ripple {...getRippleProps()} />}
 -     </Component>
 -   );
-  }

  return (
    <Component ref={domRef} className={styles} {...getButtonProps()}>
      {startContent}
      {isLoading && spinnerPlacement === "start" && spinner}
      {isLoading && isIconOnly ? null : children}
      {isLoading && spinnerPlacement === "end" && spinner}
      {endContent}
      {!disableRipple && <Ripple {...getRippleProps()} />}
    </Component>
  );

```

Reply:
Great, thank you. I corrected this